### PR TITLE
Import images script

### LINKED ISSUE #57: Implement ImportImages as scripting
I'm trying to use Image Analytics in the scripting part and I've just realized one can only import images through GUI. Would it be possible to separate the function for reading images and constructing Orange.data.Table from the widget?

e.g.
from orangecontrib.imageanalytics.images_from_file import ImportImages
images = ImportImages("../path")

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Fixes #57.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ajdapretnar** [comment]
Could you please rebase and edit commits?

**Comment by jerneju** [comment]
Of course. I squashed two commits together but this is all I can do. Three additional commits are here just because of some manoeuvre which enables us to keep a history of a copied file (`import_images.py`).

**Comment by ajdapretnar** [comment]
Is this Github's job? Keeping histories of files?

**Comment by ajdapretnar** [comment]
@inejc Could you check this, too?

File: orangecontrib/imageanalytics/import_images.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    def __init__(self, formats=DefaultFormats, report_progress=None,
+                 case_insensitive=True):
+        self.formats = formats
+        self.report_progress = report_progress
+        self.cancelled = False
+        self.case_insensitive = case_insensitive
+        self._image_meta = []
+        self._categories = {}
+
+    def _create_table(self):
```
[the remark this answers is not in the document]
Comment (jerneju) [reply orangecontrib/imageanalytics/import_images.py]: Can be a _static method_ or a function in `utils.py`?

File: orangecontrib/imageanalytics/import_images.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            meta_data = numpy.array(meta_data, dtype=object)
+
+            if len(meta_data):
+                table = Orange.data.Table.from_numpy(
+                    domain, numpy.empty((len(cat_data), 0), dtype=float),
+                    cat_data, meta_data
+                )
+            else:
+                # empty results, no images found
+                table = None
```
[the remark this answers is not in the document]
Comment (jerneju) [reply orangecontrib/imageanalytics/import_images.py]: Fixed.

File: orangecontrib/imageanalytics/import_images.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                return any(fnmatch.fnmatch(fname, pattern)
+                           for pattern in patterns)
+
+            filenames = [fname for fname in filenames
+                         if matches_any(fname, include_patterns)
+                            and not matches_any(fname, exclude_patterns)]
+
+            yield from (os.path.join(dirpath, fname) for fname in filenames)
+
+    def _image_meta_data(self, path):
```
[the remark this answers is not in the document]
Comment (jerneju) [reply orangecontrib/imageanalytics/import_images.py]: Function in `utils.py`, a static method or a function in the same file?

File: orangecontrib/imageanalytics/import_images.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ales-erjavec) [reply orangecontrib/imageanalytics/import_images.py]: A function in this module's global namespace.

File: orangecontrib/imageanalytics/import_images.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ales-erjavec) [reply orangecontrib/imageanalytics/import_images.py]: Function in the same file.

**Pull request merged**